### DISCUSSION TITLE
[Suggestion] Form generator

### DISCUSSION THREAD

**Discussion by y3lousso** [element-plus/element-plus#1617] [Ideas]
Do you plan to add a form-generator component to your framework. By that, I mean a component that can create form based on pure json.

An example for vue 2 : https://github.com/vue-generators/vue-form-generator

Since you already have already implemented all necessary fields, it would be a great addition for people who wants to generate dynamic forms based on API endpoints.

**Comment by jw-foss** [comment]
Well I think this could be some standalone feature that we could build it outside this project. Maybe we will initiate a project to do this instead of building it inside this project. This project is sort of atomic components.

**Comment by purepear** [comment]
I haven't tried it yet, but [FormVueLate](https://formvuelate.js.org/) looks promising. It's framework agnostic.
There are a bunch of other form generator libs too. [Vuetify-form-base](https://github.com/wotamann/vuetify-form-base) even has an [Element plus Demo](https://wotamann.github.io/)

**Comment by tolking** [comment]
I wrote a similar component [element-pro-components Form](https://tolking.github.io/element-pro-components/zh-CN/components/form).  

``` vue
<template>
  <pro-form  v-model="form"   :columns="columns" @submit="submit"  />
<template>
<script setup>
  const form = ref({})
  const columns = ref([
      {
        label: 'Name',
        prop: 'name',
        component: 'el-input',
      },
      {
        label: 'Address',
        prop: 'address',
        component: 'el-input',
      },
    ])
<script>
```
